opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -31,22 +38,28 @@
 
     private final AuthenticationSessionRepository authSessions;
     private final ServiceSessionRepository serviceSessions;
+    private final EncryptionService encryptionService;
 
     @Transactional
     @SneakyThrows
     public <T extends FacadeServiceableGetter> ServiceContext<T> provide(T request) {
-        if (null == request.getFacadeServiceable()) {
+        FacadeServiceableRequest facadeServiceable = request.getFacadeServiceable();
+        if (null == facadeServiceable) {
             throw new IllegalArgumentException("No serviceable body");
         }
 
         AuthSession authSession = extractAndValidateAuthSession(request);
         ServiceSession session = extractOrCreateServiceSession(request, authSession);
 
+        byte[] decryptedSecretKey = encryptionService.decryptSecretKey(session.getSecretKey());
+        byte[] decryptedData = encryptionService.decrypt(session.getContext().getBytes(), decryptedSecretKey);
+        FacadeServiceableRequest facadeServiceableDecrypted = MAPPER.readValue(decryptedData, FacadeServiceableRequest.class);
+
         return ServiceContext.<T>builder()
                 .serviceSessionId(session.getId())
                 .serviceBankProtocolId(null == authSession ? null : authSession.getParent().getProtocol().getId())
                 .authorizationBankProtocolId(null == authSession ? null : authSession.getProtocol().getId())
-                .bankId(request.getFacadeServiceable().getBankId())
+                .bankId(facadeServiceable.getBankId() == null ? facadeServiceableDecrypted.getBankId() : facadeServiceable.getBankId())
```

Comment:
Why `request.getFacadeServiceable().getBankId()` is read from facadeServiceableDecrypted ? It is used only on 1st request

Reply:
ok maybe bankId not needed, but what data will be provided from decrypted?
request.getFacadeServiceable() and facadeServiceableDecrypted should be somehow merged